### PR TITLE
logs SOL price and RSI on console from trading view page

### PR DESCRIPTION
**Pull request by shivaji43**
1. **Set RSI and 4H candles** on your TradingView account manually.
2. **Install packages** by running `yarn` in your project folder.
3. **Navigate to `src` folder** and run `node index.js`.
4. **Login manually** with the same account and complete captcha (it will wait for 2 minutes).
5. **Price and RSI** will be logged every 30 minutes.
. 
 This pull request was created for https://app.gib.work/bounties/e6c5992c-08fd-43dc-8ce8-bc4398b09145 in an attempt to solve a bounty #1 . Payment for the bounty is immediately sent to the contributor after merge.

### CODE REVIEW

**Comment by shivaji43** [comment]
@JinxArcane235 please take a look and try it for yourself 
if you need any changes , I will try to implement them quickly

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const puppeteer = require('puppeteer');
```
Comment (JinxArcane235) [inline src/index.js]: Please convert this to TypeScript. Instead of using an IIAFE, create a main function and call it explicitly

**Review by JinxArcane235 — CHANGES_REQUESTED** [review]

**Comment by JinxArcane235** [comment]
after login nothing happens

image

**Comment by shivaji43** [comment]
> after login nothing happens
> 
> image

I have put a 2 min time delay there for the login to be completed i will reduce it to 1 min.

File: src/index.js (hunk 1 above)
Comment (shivaji43) [reply src/index.js]: > Please convert this to TypeScript. Instead of using an IIAFE, create a main function and call it explicitly

Okay no problem

**Comment by shivaji43** [comment]
Screenshot 2024-11-27 073007

I have converted everything to typescript ,also  I have reduced the time for login to 1 minute and the info is logged every 1 second we can change it later on how we want it by just changing the milliseconds
(the time when `proceeding to chart after 1 minute` gets logged it will go onto solana chart and start logging automatically)


now to run 
1)`yarn`
2)`yarn dev`

**Comment by JinxArcane235** [comment]
Nice! Thanks

I just found two bugs, and I think we are done

This TradingView URL is different for me, which is: `https://www.tradingview.com/chart/hmLYX6Nq/?symbol=COINBASE%3ASOLUSD`

Also, the RSI query selector is returning a different value because the class name is the same

image


image

**Comment by shivaji43** [comment]
Screenshot 2024-11-27 182228

i cannot access this chart layout directly, you can give me access to the layout someway, and then i can change the css classname accordingly

**Comment by shivaji43** [comment]
Screenshot 2024-11-27 183825

you can Enable sharing here so i can see the chart layout

**Comment by JinxArcane235** [comment]
Enabled the sharing.

I checked here, and it seems that TradingView generates a unique url with an id for each user. I think the best approach here is to search for "SOL USD". What do you think?

**Comment by shivaji43** [comment]
> Enabled the sharing.
> 
> I checked here, and it seems that TradingView generates a unique url with an id for each user. I think the best approach here is to search for "USL USD". What do you think?

share me the link now that you have  enabled, i think that will work

**Comment by shivaji43** [comment]
Got it thank you I will fix the bugs right immediately @JinxArcane235

**Comment by JinxArcane235** [comment]
Thanks <3 @shivaji43

**Comment by shivaji43** [comment]
@JinxArcane235 you can check now , the correct rsi value is getting logged

**Comment by JinxArcane235** [comment]
Works!

**Review by JinxArcane235 — APPROVED** [review]

**Comment by shivaji43** [comment]
Thanks ♥️@JinxArcane235

**Pull request merged**